flocx_market/api/bid.py

```diff
@@ -20,6 +20,9 @@ def get(cls, marketplace_bid_id=None):
     def post(cls):
 
         data = request.get_json(force=True)
+        if type(data['server_config_query']) is not dict or \
+                data['server_config_query']['specs'] is None:
+            return {'message': 'Bad bid config.'}, 404
```

Comment:
@tzumainn Should we be performing this check in the model (like we do for `cost` in https://github.com/CCI-MOC/flocx-market/blob/master/flocx_market/db/sqlalchemy/models.py#L45)?

Reply:
Also, shouldn't `data['server_config_query']` be a list?

Reply:
Ah, yeah, this check should probably be done in the db model, or perhaps the db api.

`server_config_query` is apparently a dict that has a property named `specs` that specifies a list of matcher requirements.

Reply:
(relevant matcher code: https://github.com/CCI-MOC/flocx-market/blob/master/flocx_market/matcher/match_engine.py#L25)

Reply:
> Ah, yeah, this check should probably be done in the db model, or perhaps the db api.

...so, uh, which do you prefer? Previously we had discussed the db model being the right place to do data validation, but were you thinking that this situation is somehow different so that maybe the db api is a better place?

Reply:
Oh, I just wasn't sure how to validate this within the model itself, but I'm not suuuuper experienced with this.